snapshots/devmapper/snapshotter.go

```diff
@@ -438,11 +510,18 @@ func (s *Snapshotter) buildMounts(snap storage.Snapshot) []mount.Mount {
 	mounts := []mount.Mount{
 		{
 			Source:  s.getDevicePath(snap),
-			Type:    fsTypeExt4,
+			Type:    s.getFsType(s.getDevicePath(snap)),
```

Comment:
I'm concerned that we have to read volumes on each mount list request. Please have a look on my comment above regarding possible alternatives.

Reply:
Updated the PR implementing with labels. removed the disk reads.